functions.j2

```diff
@@ -187,13 +72,30 @@ function {{ function }}() {
   done
 {%- endif %}
 
+  easy_infra_{{ function | replace("-", "_") }}_hooks=()
+  if [[ "${DISABLE_HOOKS:-false}" == "true" ]]; then
+    _feedback WARNING "DISABLE_HOOKS is set to ${DISABLE_HOOKS}; skipping all hooks for the {{ function }} command..."
+    _log "easy_infra.stdouterr" info unknown "{{ function }}" "${dir}" string "The DISABLE_HOOKS environment variable is set to ${DISABLE_HOOKS}, skipping all hooks for the {{ function }} command"
+  else
+    # Dynamically register hooks
+    for file in /opt/hooks/bin/*.sh; do
+      command=$(awk -F\: '/register_hook/ { gsub(/ /,""); print $2 }' "${file}")
+      if [[ "${command}" == "{{ function }}" ]]; then
+        easy_infra_{{ function | replace("-", "_") }}_hooks+=("${file}")
+      fi
+    done
+  fi
+
   # Methods to skip **all** security tools for {{ function }}
   DISABLE_SECURITY="${DISABLE_SECURITY:-false}"
   dirs=()
   if [[ "${AUTODETECT:-false}" == "true" ]]; then
 {%- if file_extensions is defined %}
 {%- for file_extension in file_extensions %}
-    IFS=$'\n' dirs+=($(find . -iname "*.{{ file_extension }}" -type f -exec dirname {} \; | sort -u | xargs readlink --canonicalize))
+    files=$(find . -iname "*.{{ file_extension }}" -type f)
```

Comment:
This avoids an ambiguous error message of `readlink: missing operand` when no files are detected with the given file extension.